QuickOSM/core/style.py

```diff
@@ -0,0 +1,181 @@
+""" Build a qml file with the field colour from OSM """
+
+import io
+
+from QuickOSM.core.utilities.tools import style_color_file
+
+__copyright__ = 'Copyright 2021, 3Liz'
+__license__ = 'GPL version 3'
+__email__ = 'info@3liz.org'
+
+
+def write_style_color_line(colors: list) -> str:
```

Comment:
Nice idea.
But this will be difficult to maintain in the future. (if the XML is changing etc).

You can use straight the QGIS API to set the style.
No need to generate the XML and then apply it using PyQGIS.
Load straight from PyQGIS the style.